packages/aws-cdk/lib/api/deployments.ts

```diff
@@ -417,6 +493,120 @@ export class Deployments {
     });
   }
 
+  public async rollbackStack(options: RollbackStackOptions): Promise<RollbackStackResult> {
+    const {
+      stackSdk,
+      resolvedEnvironment: _,
+      cloudFormationRoleArn,
+      envResources,
+    } = await this.prepareSdkFor(options.stack, options.roleArn, Mode.ForWriting);
+
+    if (options.validateBootstrapStackVersion ?? true) {
+      // Do a verification of the bootstrap stack version
+      await this.validateBootstrapStackVersion(
+        options.stack.stackName,
+        BOOTSTRAP_STACK_VERSION_FOR_ROLLBACK,
+        options.stack.bootstrapStackVersionSsmParameter,
+        envResources);
+    }
+
+    const cfn = stackSdk.cloudFormation();
+    const deployName = options.stack.stackName;
+
+    while (true) {
+      let cloudFormationStack = await CloudFormationStack.lookup(cfn, deployName);
+
+      // TODO: currently, we only roll back UPDATE_FAILED and CREATE_FAILED stacks. Conceivably, we could also
+      // handle UPDATE_ROLLBACK_FAILED by offering some option like "force rollback".
+      switch (cloudFormationStack.stackStatus.rollbackChoice) {
+        case RollbackChoice.NONE:
+          warning(`Stack ${deployName} does not need a rollback: ${cloudFormationStack.stackStatus}`);
```

Comment:
The message is misleading, it might be that the stack cannot be rolled back, in which case I think we should error out because we didn't get the user to their desired state.

Reply:
This was for the case where the app has multiple stacks. We shouldn't require that ALL of them are rollbackable, we just want to get back to a stable state.

But this reminded me that I had the plan to add something to the driving loop that I forgot, so I'll add the condition there that at least one stack needs to be in rollbackable state.

Reply:
I see. You would prefer `throw new NotRollbackeableError()` and then catch specifically that, so that this function can happily fail